swan-lake/development-tutorials/additional-tools/scan-tool.md

```diff
@@ -85,6 +85,98 @@ To ignore a specific set of rules during the analysis, use the following command
+- [SonarQube Ballerina plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580146), and [SonarQube platform plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580149) downloaded.
+
+### Configure the SonarQube server
+
+1. Install [Java 17](https://adoptium.net/en-gb/temurin/releases/?version=17) in your machine.
+   - If Java 17 is not the default Java installation, override it:
+
+      **For Unix/macOS:**
+      ```bash
+      export SONAR_JAVA_PATH="path/to/java17_home/bin/java"
+      ```
+
+      **For Windows:**
+      ```bash
+      setx SONAR_JAVA_PATH="path\to\java17_home\bin\java"
+      ```
+
+2. Setup SonarQube 9.9 LTA
+   - Download SonarQube 9.9 LTA from [here](https://www.sonarsource.com/products/sonarqube/downloads/historical-downloads/).
+   - Extract the downloaded zip file.
+
+3. Add the [SonarQube Ballerina plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580146).
+   - Download the latest Ballerina SonarQube plugin JAR.
+   - Place the JAR file into the `extensions/plugins/` directory of your SonarQube installation.
+
+4. Navigate to the appropriate `bin/<OS>/` directory and run:
+   ```bash
+   $ ./sonar.sh start
+   ```
+
+   You can access the SonarQube dashboard at http://localhost:9000 once the server is up.
+
+5. Create a new project in SonarQube.
+   - Log in to the SonarQube dashboard.
+   - Click on `Create Project`.
+   - Follow the prompts to set up your project.
+
+6. Install and Configure SonarScanner CLI.
+   - Download SonarScanner CLI from [here](https://docs.sonarsource.com/sonarqube-server/9.9/analyzing-source-code/scanners/sonarscanner/).
+   - Add it to your system `PATH`.
+   - Ensure `sonar.host.url` is set correctly (either via a properties file or CLI parameter).
+
+### Configure the Ballerina project
+
+1. Download the [SonarQube platform plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580149).
+
+2. Create a `sonar-project.properties` file at the root of your Ballerina project with the following content:
+   ```properties
+   sonar.projectKey=<your-project-key>
+   sonar.projectName=<your-project-name>
+   ```
+
+3. Create a Scan.toml at the root of your Ballerina project. Add Additional SonarQube Configurations by referencing the `sonar-project.properties` file.
+   ```toml
+   [[platform]]
+   name = "sonarqube"
+   path = "<path-to-sonar-platform-plugin>"
+   sonarProjectPropertiesPath = "<path-to-sonar-project.properties>"
+   ```
+
+### Publish reports to SonarQube
+
+1. Link a ballerina source repo to the SonarQube server from a DevOps platform or manually.
+
+2. Authenticate Using a Token.
+   - Generate a token from the `My Account`->`Security` section in the SonarQube UI.
+   - Set the token as an environment variable:
+
+      **For Unix/macOS:**
+      ```bash
+      $ export SONAR_TOKEN=<your-token>
+      ```
+
+      **For Windows:**
+      ```bash
```

Comment:
Remove the `bash` keyword.

Reply:
Fixed with 56a4413a009690234ac0707f58289ea3558b593e